### DISCUSSION TITLE
Jobs Queue system instable and lacks proper documentation

### DISCUSSION THREAD

**Discussion by yonnic** [payloadcms/payload#9418] [General]
hey guys, for 3 days i am trying to get this Jobs Queue system to work but the instability and lack of documentation is a huge blocker for me - thinking about to ditch it completely.

Here is whats wrong about it.
* It is not clear where to put "waitUntil" It was mentioned once in the docs but its not clear how to use it.
* Calling `/api/payload-jobs/run` never works for me, i always get "No jobs to run".
* Running `npx payload jobs:run --queue default` randomly works sometimes but get stuck after a response, or it process for 20 sec and returns nothing.
* `deleteJobOnComplete: false,` seems to have no effect, once a job was run its gone - how to do cron then ? 

There are so many other things missing in the docs or as features. 
* How to list jobs ? 
* How to delete jobs ?
* How to monitor or check "processing" jobs (saw it in the response on job creation)

I really like the concept of this system and how its suppose to work, but at current state it has too many flaws and is unusable sadly. Maybe something is messed up on my instance but everything else works perfectly fine.

Payload 3.0.2
Node v20.15.1
Mac OS Sequoia 15.2

**Comment by yonnic** [comment]
seems like no one actually tried or use the Jobs system, will check if i create an issue here for every bug i had with it

**Comment by DanRibbens** [comment]
We just merged https://github.com/payloadcms/payload/pull/9283 that will properly delete completed jobs.

**Comment by DanRibbens** [comment]
> Calling /api/payload-jobs/run never works for me, i always get "No jobs to run".

This was addressed https://github.com/payloadcms/payload/pull/9509 released [v3.1.1](https://github.com/payloadcms/payload/releases/tag/v3.1.1)

> Running npx payload jobs:run --queue default randomly works sometimes but get stuck after a response, or it process for 20 sec and returns nothing.

This might have been because of an issue we found and patched in [v3.0.1](https://github.com/payloadcms/payload/releases/tag/v3.0.1).

We will continue to improve the jobs queue and the docs and appreciate your feedback!

**Comment by MinskLeo** [comment]
Hi! Would be cool to actually see the example of scheduling using `waitUntil`. I agree - thats very unclear where the property should be set.

**Comment by nightwalker89** [comment]
Payload 3.36.1
I add try catch in task handler but the failed job still make the whole app crash.

**Comment by mariovalney** [comment]
Payload 3.53.0
Jobs with {"cancelled": true} are not deleted even without "deleteJobOnComplete".

Would be cool to kept logs or we should keep "deleteJobOnComplete" true to have historic data?

**Comment by Caoska** [comment]
Totally echoing this — some queue APIs give you the building blocks but not much help in understanding what’s actually running or when.

When you bump into instability like this, do you tend to build your own observability layer, or lean on UIs/dashboards like Bull Board, or something else?
